platform/x11/os_x11.cpp

```diff
@@ -1885,12 +1885,23 @@ void OS_X11::process_xevents() {
 						// Determine the axis used (called valuators in XInput for some forsaken reason)
 						//  Mask is a bitmask indicating which axes are involved.
 						//  We are interested in the values of axes 0 and 1.
-						if (raw_event->valuators.mask_len <= 0 || !XIMaskIsSet(raw_event->valuators.mask, 0) || !XIMaskIsSet(raw_event->valuators.mask, 1)) {
+						if (raw_event->valuators.mask_len <= 0) {
 							break;
 						}
 
-						double rel_x = raw_event->raw_values[0];
-						double rel_y = raw_event->raw_values[1];
+						const double *values = raw_event->raw_values;
```

Comment:
Is this additional pointer necessary? Otherwise this seems more readable to me:
```
if (XIMaskIsSet(raw_event->valuators.mask, 0)) {
	rel_x = raw_event->raw_values[0];
}
if (XIMaskIsSet(raw_event->valuators.mask, 1)) {
	rel_y = raw_event->raw_values[1];
}
```

Reply:
The issue is that when mask 0 isn't set, rel_y should actually be `raw_values[0]`
So this:
```
const double *values = raw_event->raw_values;

if (XIMaskIsSet(raw_event->valuators.mask, 0)) {
    rel_x = *values;
    values++;
}

if (XIMaskIsSet(raw_event->valuators.mask, 1)) {
    rel_y = *values;
}
```

Is equivalent to:
```
if (XIMaskIsSet(raw_event->valuators.mask, 0)) {
    rel_x = raw_event->raw_values[0];

    if (XIMaskIsSet(raw_event->valuators.mask, 1)) {
        rel_y = raw_event->raw_values[1];
    }
} else if (XIMaskIsSet(raw_event->valuators.mask, 1)) {
    rel_y = raw_event->raw_values[0];
}
```
I'll defer to your judgement on that one. Found it by looking at glfw's implementation.
https://github.com/glfw/glfw/blob/62993d9391469e87c3c69592d436122d73db8b07/src/x11_window.c#L1202

Reply:
Oh indeed, that makes sense how you did it.